### PR TITLE
Adds mcrypt extension for PHP 5.6.40

### PR DESCRIPTION
**Pull request by nickkadutskyi**
Adds missing mcrypt ext from php-src.

### CODE REVIEW

**Comment by nick-kadutskyi** [comment]
Fixes this issue https://github.com/fossar/nix-phps/issues/337

File: pkgs/package-overrides.nix [hunk 1]
```diff
@@ -43,6 +43,35 @@ in
...
+        if lib.versionOlder prev.php.version "7.0" then
+          prev.mkExtension {
+            name = "mcrypt";
+            version = version;
+            src = pkgs.fetchFromGitHub {
+              owner = "php";
+              repo = "php-src";
+              rev = revision;
+              sha256 = "sha256-aDiGbOqy2L8Qqd6IliEN19Oa8JO1uDSEju6uUN4b0XI=";
+            } + "/ext/mcrypt";
```
Comment (jtojnar) [inline pkgs/package-overrides.nix:61]: Why not use `prev.php.src` pulled in by `mkExtension` by default?

```suggestion
```

File: pkgs/package-overrides.nix [hunk 2]
```diff
@@ -43,6 +43,35 @@ in
...
+            nativeBuildInputs = [ pkgs.pkg-config ];
+            buildInputs = [ pkgs.libmcrypt ];
+            configureFlags = [ "--with-mcrypt=${pkgs.libmcrypt.outPath}" ];
+            doCheck = true;
+            checkTarget = "test";
```
Comment (jtojnar) [inline pkgs/package-overrides.nix:66]: This is already hardcoded by `mkExtension`.

File: pkgs/package-overrides.nix (hunk 1 above)
Comment (nick-kadutskyi) [reply pkgs/package-overrides.nix]: I replaced it with prev.php.src in next commit

File: pkgs/package-overrides.nix (hunk 2 above)
Comment (nick-kadutskyi) [reply pkgs/package-overrides.nix]: I removed everything except `--with-mcrypt=` config flag. It doesn't work without that flag.

File: pkgs/package-overrides.nix (hunk 1 above)
Comment (jtojnar) [reply pkgs/package-overrides.nix]: Here are examples of what to pass to `mkExtension`: https://github.com/NixOS/nixpkgs/blob/3f01f05941eec3a87bbf5baae8edaea6ce3615fa/pkgs/top-level/php-packages.nix#L360

Note that `src` and build directory are already properly set based on `name`:

https://github.com/NixOS/nixpkgs/blob/3f01f05941eec3a87bbf5baae8edaea6ce3615fa/pkgs/top-level/php-packages.nix#L129

File: pkgs/package-overrides.nix [hunk 3]
```diff
@@ -43,6 +43,19 @@ in
   };
 
   extensions = prev.extensions // {
+
+    mcrypt =
```
Comment (jtojnar) [inline pkgs/package-overrides.nix:47]: Also, please insert new extensions in alphabetically correct order.

File: pkgs/package-overrides.nix (hunk 2 above)
Comment (jtojnar) [reply pkgs/package-overrides.nix]: `src` is still there though.

File: pkgs/package-overrides.nix (hunk 1 above)
Comment (nick-kadutskyi) [reply pkgs/package-overrides.nix]: Cleaned it again. Thanks for the references, I didn't know exactly how it handle that.

File: pkgs/package-overrides.nix (hunk 2 above)
Comment (nick-kadutskyi) [reply pkgs/package-overrides.nix]: Removed src

File: pkgs/package-overrides.nix
Comment (jtojnar) [inline pkgs/package-overrides.nix:1]: Oh, and we will also want it to have it run by CI:

https://github.com/fossar/nix-phps/blob/d40b1a19f521b1c495f60af2a9ac40e5c27d0925/.github/workflows/build.yaml#L78-L80

**Review by jtojnar — COMMENTED** [review]
Also when ready, please combine the changes into a single commit titled `php56.extensions.mcrypt: init`

**Comment by nick-kadutskyi** [comment]
> Also when ready, please combine the changes into a single commit titled `php56.extensions.mcrypt: init`

Just force-pushed the squash

**Comment by jtojnar** [comment]
Looks like it also needs a check definition https://github.com/fossar/nix-phps/blob/d40b1a19f521b1c495f60af2a9ac40e5c27d0925/checks.nix#L32-L36

**Comment by nickkadutskyi** [comment]
> Looks like it also needs a check definition
> 
> https://github.com/fossar/nix-phps/blob/d40b1a19f521b1c495f60af2a9ac40e5c27d0925/checks.nix#L32-L36

Just add `mcrypt` into checks.

**Review by drupol — APPROVED** [review]

**Comment by jtojnar** [comment]
Thanks.

**Pull request merged**